code/datums/controllers/explosion_controls.dm

```diff
@@ -216,11 +217,22 @@ var/datum/explosion_controller/explosions
 			E.set_up(epicenter)
 			E.start()
 
-		var/radius = round(sqrt(power), 1) * brisance * range_cutoff_fraction
-
 		if (istype(source)) // Cannot read null.fingerprintslast
 			last_touched = source.fingerprintslast
 
+			if(istype(source, /obj/item))
+				var/list/covering_mobs = list()
+				for (var/mob/living/carbon/human/H in epicenter)
+					if(H.lying && H.hasStatus("blocking")) //trying to cover it
+						covering_mobs.Add(H)
+				var/mob/living/carbon/human/hero = PICK_OR_NULL(covering_mobs)
+				if(hero) //there goes my hero
+					hero.visible_message("<span class='alert'>[hero] wraps [his_or_her(hero)] body tightly around [src]!!</span>")
+					hero.cover_explosion_with_body(power, brisance)
+					power *= ITEM_EXPLOSION_COVERED_REDUCTION_MULT
```

Comment:
would make more sense for `cover_explosion_with_body()` to return a modified `power` imo, and/or have a maximum amount that power can be lowered by - e.g.: a maxcap TTV should probably be relatively unaffected